pkg/controller/aws/network/routetable/controller.go

```diff
@@ -0,0 +1,307 @@
+	cr, ok := mgd.(*v1alpha1.RouteTable)
+	if !ok {
+		return resource.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	// To find out whether a RouteTable exist:
+	// - the object's ExternalState should have routeTableId populated
+	// - a RouteTable with the given routeTableId should exist
+	if cr.Status.RouteTableID == "" {
+		return resource.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	req := e.client.DescribeRouteTablesRequest(&awsec2.DescribeRouteTablesInput{
+		RouteTableIds: []string{cr.Status.RouteTableID},
+	})
+	req.SetContext(ctx)
+
+	response, err := req.Send()
+
+	if err != nil {
+		if ec2.IsRouteTableNotFoundErr(err) {
+			return resource.ExternalObservation{
+				ResourceExists: false,
+			}, nil
+		}
+
+		return resource.ExternalObservation{}, errors.Wrapf(err, "failed to describe RouteTable with id: %v", cr.Status.RouteTableID)
+	}
+
+	// in a successful response, there should be one and only one object
+	if len(response.RouteTables) != 1 {
+		return resource.ExternalObservation{}, errors.Errorf("retrieved multiple RouteTables for the given routeTableId: %v", cr.Status.RouteTableID)
+	}
+
+	observed := response.RouteTables[0]
+
+	stateAvailable := true
+	for _, rt := range observed.Routes {
+		if rt.State != awsec2.RouteStateActive {
+			stateAvailable = false
+			break
+		}
+	}
+	if stateAvailable {
+		cr.SetConditions(runtimev1alpha1.Available())
+	}
+
+	cr.Status.RouteTableExternalStatus = cr.BuildExternalStatusFromObservation(observed)
+
+	return resource.ExternalObservation{
+		ResourceExists:    true,
+		ConnectionDetails: resource.ConnectionDetails{},
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (resource.ExternalCreation, error) { // nolint:gocyclo
+	cr, ok := mgd.(*v1alpha1.RouteTable)
+	if !ok {
+		return resource.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	req := e.client.CreateRouteTableRequest(&awsec2.CreateRouteTableInput{
+		VpcId: aws.String(cr.Spec.VPCID),
+	})
+	req.SetContext(ctx)
+	result, err := req.Send()
+	if err != nil {
+		return resource.ExternalCreation{}, errors.Wrap(err, "failed to create the RouteTable resource")
+	}
+
+	cr.Status.RouteTableExternalStatus = cr.BuildExternalStatusFromObservation(*result.RouteTable)
+
+	// Create Routes
+	if err := e.createRoutes(ctx, cr.Status.RouteTableID, cr.Spec.Routes, cr.Status.Routes); err != nil {
+		return resource.ExternalCreation{}, err
+	}
+
+	// Create Associations
+	if err := e.createAssociations(ctx, cr.Status.RouteTableID, cr.Spec.Associations, cr.Status.Associations); err != nil {
+		return resource.ExternalCreation{}, err
+	}
+
+	return resource.ExternalCreation{}, nil
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (resource.ExternalUpdate, error) {
+	// TODO(soorena776): add more sophisticated Update logic, once we
+	// categorize immutable vs mutable fields (see #727)
```

Comment:
Is the intention to add this Update functionality in a future PR?

Reply:
Yes. Once we resolved #727